### ISSUE TITLE
NSFW images are visible before being replaced

### ISSUE THREAD

**Issue by kantaiz3** [silveredgold/beta-protection#12]
Hello! I just discovered your extension and will be switching to it from now on. The extensive customizability won me over :) However, I discovered a few things that could be improved, so I'll open issues with bugs and suggestions.

NSFW images are visible for almost second before being replaced by a placeholder image. This seems to happen every time an image is loaded.

BetaSafety doesn't have this problem, so we might want to be inspired by what they are doing ;)

**Comment by silveredgold** [comment]
Is this when viewing an image directly (i.e. not part of a web page)?

If so, that's a browser limitation and I haven't found a way around that that will still work after the MV2 deprecation in March.

**Comment by kantaiz3** [comment]
> Is this when viewing an image directly

Yes, but it also happens on some regular websites. I noticed this on rule34.

> If so, that's a browser limitation and I haven't found a way around

Hmmm I don't know the specifics, but couldn't we [hide or blur out](https://developer.mozilla.org/en-US/docs/Web/CSS/filter) all `<img>`s until they are replaced by a placeholder or the censored image? If we could insert styles into the page fast enough, this might help.

**Comment by silveredgold** [comment]
Hiding an `<img>` won't work when the image _is_ the page, because there's no `<img>` to hide. Also, hiding `<img>` is how it already works, but that's only part of it since you also need to hide all the other places that images can appear (like CSS `background-image` rules)
